_source/_posts/2021-09-12-spring-boot-spa.md

```diff
@@ -0,0 +1,686 @@
+
+If the preflight header is not handled correctly, CORS will return an error to the client application. If the preflight header does not allow the type of request or does not allow requests from the client origin, CORS will also return an error to the client. The actual, original request to the server will never be made. Only once the preflight request has been properly handled will the browser allow the request from the client app to the resource server proceed.
+
+Here is a general outline of a CORS request:
+
+- Javascript client app makes request to resource server on different domain
+- Browser intercepts request, flags it as requiring CORS verification, and sends the CORS preflight request to the resource server
+- Server responds to the preflight request by saying it will allow the client domain to make the type of request
+- Browser sends the origin request to the server
+- Server handles original request, authenticating and authorizing it using OAuth 2.0 an OIDC before returning a reply
+
+From a developer perspective, CORS is something that needs to be configured on the server. Spring Boot makes this easy. There are two places in the Java code where CORS is configured. CORS must generally be enabled for the Spring Boot application. This is done in the `SecurityConfiguration` class in the `configure(HttpSecurity http)` method by adding the `cors()` method to the `http` configuration chain. 
+
+The second place is the `@CrossOrigin` annotation on the `CaffeineLevelRestController` class. This tells Spring Boot to configure CORS for all endpoints in this controller. By default, this will allow cross-origin requests from any domain. To restrict requests to only our client application domain, we could have used the annotation `@CrossOrigin(origins = "http://localhost:8080")`.
+
+Take a look at [this Spring blog post](https://spring.io/blog/2015/06/08/cors-support-in-spring-framework) for more info on Spring Boot and CORS.
+
+## Use The Okta CLI To Configure The Resource Server
+
+You should already have used the Okta CLI to either register a new account or log into an existing account. If not, please do so now. If you type `okta login`, you should see a message that says something like: `Okta Org already configured: https://dev-123456.okta.com/`.
+
+You are going to use the CLI to configure the Spring Boot project to use Okta for JWT authentication. The CLI will create an OpenID Connect (OIDC) application on Okta's servers that will be the authentication and authorization provider for the resource server.
+
+To start the configuration process, use the following command.
+
+```bash
+okta apps create
```

Comment:
You should be able to replace this whole section with a CLI include. https://developer.okta.com/blog/setup